app/controllers/puppetclasses_controller.rb

```diff
@@ -9,9 +9,7 @@ class PuppetclassesController < ApplicationController
   before_action :setup_search_options, :only => :index
 
   def index
-    eager_load_tables = [:config_group_classes, :class_params, :environments,
-                         :hostgroups]
```

Comment:
I think `:hostgroups` is worth keeping in here. Can you check there's no performance impact for you with it?

The others definitely seem to have a negative effect on performance as the eagerly loaded versions aren't used (e.g. the info about class_params isn't used, only a count), so this is otherwise correct.